### PR TITLE
Add envoy config route validations for emoticons and substrings

### PR DESCRIPTION
**Pull request by Drifftr**
### Purpose
Add further validations for emoticons and substrings

### Issues
https://github.com/wso2-enterprise/choreo/issues/28819

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments
Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/routes_with_clusters.go [hunk 1]
```diff
@@ -1707,3 +1731,33 @@ func addSubscriptionRatelimitActions(actions []*routev3.RateLimit) []*routev3.Ra
...
+	if smileyValidated && subValidated {
+		return nil
+	} else if !smileyValidated {
+		return errors.New("route path contains an emoticon")
+	}
+	return errors.New("route path contains a substitution string")
+}
+
+// Validate envoy route method whether its of known type
+func validateEnvoyRouteMethodString(method string) error {
```
Comment (VirajSalaka) [inline adapter/internal/oasparser/envoyconf/routes_with_clusters.go:1756]: From enforcer what we support is anyway "GET", "POST", "PUT", "DELETE", "PATCH", "HEAD", "OPTIONS". 
Shall we remove the rest

Comment (Drifftr) [reply adapter/internal/oasparser/envoyconf/routes_with_clusters.go:1756]: Thanks, Fixed.

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20240813.19

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240813.19

**Comment by choreo-cicd** [comment]
[succeeded]  : dev-deployment-v2 : 20240813.19

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240815.4

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20240815.3

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240815.5

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240815.5

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240815.5

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240815.5

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20240816.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240816.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240820.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240820.3

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240820.3

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20240820.3

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240820.4

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : prod-deployment-v2 : 20240820.4

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240820.4

**Pull request merged**